corehq/apps/export/models/new.py

```diff
@@ -0,0 +1,370 @@
+        return item
+
+
+class ExportColumn(DocumentSchema):
+    item = SchemaProperty(ExportItem)
+    label = StringProperty()
+
+    def get_value(self, doc, base_path):
+        """
+        Get the value of self.item of the given doc.
+        When base_path is [], doc is a form submission or case,
+        when base_path is non empty, doc is a repeat group from a form submission.
+        :param doc: A form submission or instance of a repeat group in a submission or case
+        :param base_path:
+        :return:
+        """
+        # Confirm the ExportItem's path starts with the base_path
+        assert base_path == self.item.path[:len(base_path)]
+        # Get the path from the doc root to the desired ExportItem
+        path = self.item.path[len(base_path):]
+        return NestedDictGetter(path)(doc)
+
+
+class TableConfiguration(DocumentSchema):
+    name = StringProperty()
+    repeat_path = ListProperty()
+    columns = ListProperty(ExportColumn)
+
+    def get_rows(self, document):
+        """
+        Return a list of ExportRows generated for the given document.
+        :param document: dictionary representation of a form submission or case
+        :return: List of ExportRows
+        """
+        # Note that sub_documents will be [document] if self.repeat_path is []
+        sub_documents = self._get_sub_documents(self.repeat_path, [document])
+        rows = []
+        for doc in sub_documents:
+
+            row_data = []
+            for col in self.columns:
+                val = col.get_value(doc, self.repeat_path)
+                if isinstance(val, list):
+                    row_data.extend(val)
+                else:
+                    row_data.append(val)
+            rows.append(ExportRow(data=row_data))
+        return rows
+
+    def _get_sub_documents(self, path, docs):
+        """
+        Return each instance of a repeat group at the path from the given docs.
+        If path is [], just return the docs
+
+        >>> TableConfiguration()._get_sub_documents(['foo'], [{'foo': {'bar': 'a'}}, {'foo': {'bar': 'b'}}])
+        [{'bar': 'a'}, {'bar': 'b'}]
+        >>> TableConfiguration()._get_sub_documents(['foo', 'bar'], [{'foo': [{'bar': {'baz': 'a'}}, {'bar': {'baz': 'b'}},]}]
+        [{'baz': 'a'}, {'baz': 'b'}]
+
+        :param path: A list of a strings
+        :param docs: A list of dicts representing form submissions
+        :return:
+        """
+        if len(path) == 0:
+            return docs
+
+        new_docs = []
+        for doc in docs:
+            next_doc = doc.get(path[0], {})
+            if type(next_doc) == list:
+                new_docs.extend(next_doc)
+            else:
+                new_docs.append(next_doc)
+        return self._get_sub_documents(path[1:], new_docs)
+
+
+class ExportInstance(Document):
+    tables = ListProperty(TableConfiguration)
+
+    class Meta:
+        app_label = 'export'
+
+
+class ExportRow(object):
```

Comment:
I was imagining this class would do something like [`FormattedRow`](https://github.com/dimagi/commcare-hq/blob/04257f5864e3f1d44cbc95e85080369358b868a2/corehq/ex-submodules/couchexport/export.py#L346), but wasn't sure yet what other properties would be needed.